File: leetcode-submissions/142_Linked_List_Cycle_II.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    ListNode *detectCycle(ListNode *head) {
        ListNode * pos=new ListNode(-1);
        if(head==NULL or head->next==NULL){
            return pos;
        }
      ListNode * slow=head;
      ListNode * fast=head->next;
      while(fast and fast->next){
          if(slow==fast){
              break;
          }
          slow=slow->next;
          fast=slow->next;
      }  
      fast=head;
      int p=0;
      while(fast->next!=slow){
          p++;
          fast=fast->next;
      }
      ListNode * i=new ListNode(p);
      return i;
    }
};

File: leetcode-submissions/142_Linked_List_Cycle_II.ts
/**
 * Definition for singly-linked list.
 * class ListNode {
 *     val: number
 *     next: ListNode | null
 *     constructor(val?: number, next?: ListNode | null) {
 *         this.val = (val===undefined ? 0 : val)
 *         this.next = (next===undefined ? null : next)
 *     }
 * }
 */

function detectCycle(head: ListNode | null): ListNode | null {
    if(head===null || head.next===null){
        return null;
    }
    let slow:ListNode=head;
    let fast:ListNode=head;
    while(fast && fast.next){
        slow=slow.next;
        fast=fast.next.next;
        if(slow===fast){
            fast=head;
            while(slow!==fast){
                slow=slow.next;
                fast=fast.next;
            }
            return slow;
        }
    }
return null;
};